src/JsonSchema/SchemaStorage.php

```diff
@@ -27,29 +31,13 @@ public function getUriRetriever()
 	}
 
 	/**
-	 * @param UriRetrieverInterface $uriRetriever
-	 */
-	public function setUriRetriever($uriRetriever)
```

Comment:
This approach prevents reuse of an instance of `SchemaStorage` by only allowing a user to set the retriever or resolver only during instantiation. The goal of simplifying instantiation `SchemaStorage` is achieved but I am not sure that is a problem nor am I sure it is germane to the main goal of handling circular references.

I am not sure I think this is an improvement but I won't hold up this PR for this if the community supports this move.